[Automation] javac compile fails for org.asynchttpclient:async-http-client:3.0.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25296069081/job/74155119094
Last 300 lines of the log:
```console
org_asynchttpclient.async_http_client.NamePasswordCallbackHandlerTest > invokesGenericPasswordCallbackMethod() SUCCESSFUL

org_asynchttpclient.async_http_client.ReleasePermitOnCompleteTest > releasesPermitAfterThrowableNotification() SUCCESSFUL

org_asynchttpclient.async_http_client.ReleasePermitOnCompleteTest > releasesPermitAfterSuccessfulCompletion() SUCCESSFUL


Test run finished after 9 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[         4 tests found           ]
[         0 tests skipped         ]
[         4 tests started         ]
[         0 tests aborted         ]
[         4 tests successful      ]
[         0 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 18s
18 actionable tasks: 8 executed, 10 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
-------
Test for org.asynchttpclient:async-http-client:2.12.3 passed.
====================

> Task :test

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 20s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.14.5 at 2026-05-04T01:09:49Z with exit code 0 after 80s
PASSED:2.14.5
========================================================================================
 Testing org.asynchttpclient:async-http-client:2.12.3:3.0.0
========================================================================================
========================================================================================
 org.asynchttpclient:async-http-client:2.12.3:3.0.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 3.0.0 at 2026-05-04T01:09:50Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean test -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 660ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest FAILED
====================
Testing library: org.asynchttpclient:async-http-client:2.12.3
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :jar
> Task :tck-test-support:jar UP-TO-DATE
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                .setShutdownQuietPeriod(0)
                                        ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                  .setShutdownQuietPeriod(0)
                                          ^
  Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.asynchttpclient:async-http-client:2.12.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.0.0 at 2026-05-04T01:09:53Z with exit code 1 after 3s
❌ ERROR: 'native-image run' for 3.0.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean test -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"]
========================================================================================
 org.asynchttpclient:async-http-client:2.12.3:3.0.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.0.0 at 2026-05-04T01:09:53Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 700ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                .setShutdownQuietPeriod(0)
                                        ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                  .setShutdownQuietPeriod(0)
                                          ^
  Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 870ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.0.0 at 2026-05-04T01:09:56Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.0.0 failed with exit code 1.
FAILED[javac compile][3.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25296069081/job/74155119083
Last 300 lines of the log:
```console
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

org_asynchttpclient.async_http_client.ReleasePermitOnCompleteTest > releasesPermitAfterThrowableNotification() SUCCESSFUL

org_asynchttpclient.async_http_client.ReleasePermitOnCompleteTest > releasesPermitAfterSuccessfulCompletion() SUCCESSFUL


Test run finished after 6 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[         4 tests found           ]
[         0 tests skipped         ]
[         4 tests started         ]
[         0 tests aborted         ]
[         4 tests successful      ]
[         0 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 29s
18 actionable tasks: 8 executed, 10 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
-------
Test for org.asynchttpclient:async-http-client:2.12.3 passed.
====================

> Task :test

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 31s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.14.5 at 2026-05-04T01:10:00Z with exit code 0 after 92s
PASSED:2.14.5
========================================================================================
 Testing org.asynchttpclient:async-http-client:2.12.3:3.0.0
========================================================================================
========================================================================================
 org.asynchttpclient:async-http-client:2.12.3:3.0.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 3.0.0 at 2026-05-04T01:10:00Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean test -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 777ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest FAILED
====================
Testing library: org.asynchttpclient:async-http-client:2.12.3
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration

                .setShutdownQuietPeriod(0)
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html
                                        ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                  .setShutdownQuietPeriod(0)
                                          ^
  Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1s
11 actionable tasks: 1 executed, 10 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.asynchttpclient:async-http-client:2.12.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.0.0 at 2026-05-04T01:10:04Z with exit code 1 after 4s
❌ ERROR: 'native-image run' for 3.0.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean test -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"]
========================================================================================
 org.asynchttpclient:async-http-client:2.12.3:3.0.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.0.0 at 2026-05-04T01:10:04Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 793ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date
                .setShutdownQuietPeriod(0)
                                        ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                  .setShutdownQuietPeriod(0)
                                          ^
  Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 951ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.0.0 at 2026-05-04T01:10:06Z with exit code 1 after 2s
❌ ERROR: 'javac compile' for 3.0.0 failed with exit code 1.
FAILED[javac compile][3.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25296069081/job/74155119099
Last 300 lines of the log:
```console

org_asynchttpclient.async_http_client.ReleasePermitOnCompleteTest > releasesPermitAfterSuccessfulCompletion() SUCCESSFUL


Test run finished after 8 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[         4 tests found           ]
[         0 tests skipped         ]
[         4 tests started         ]
[         0 tests aborted         ]
[         4 tests successful      ]
[         0 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 36s
18 actionable tasks: 8 executed, 10 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
-------
Test for org.asynchttpclient:async-http-client:2.12.3 passed.
====================

> Task :test

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 38s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.14.5 at 2026-05-04T01:10:09Z with exit code 0 after 99s
PASSED:2.14.5
========================================================================================
 Testing org.asynchttpclient:async-http-client:2.12.3:3.0.0
========================================================================================
========================================================================================
 org.asynchttpclient:async-http-client:2.12.3:3.0.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 3.0.0 at 2026-05-04T01:10:09Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0.0" ./gradlew clean test -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 901ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.asynchttpclient:async-http-client:2.12.3
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:jar UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                .setShutdownQuietPeriod(0)
                                        ^

Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                  .setShutdownQuietPeriod(0)
                                          ^
  Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

BUILD FAILED in 1s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.asynchttpclient:async-http-client:2.12.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.0.0 at 2026-05-04T01:10:13Z with exit code 1 after 4s
❌ ERROR: 'native-image run' for 3.0.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.0.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0.0" ./gradlew clean test -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"]
========================================================================================
 org.asynchttpclient:async-http-client:2.12.3:3.0.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.0.0 at 2026-05-04T01:10:13Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 807ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED

/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                .setShutdownQuietPeriod(0)
                                        ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error

FAILURE: Build failed with an exception.

* What went wrong:
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.asynchttpclient/async-http-client/2.12.3/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java:30: error: incompatible types: int cannot be converted to Duration
                  .setShutdownQuietPeriod(0)
                                          ^
  Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 967ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.0.0 at 2026-05-04T01:10:16Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.0.0 failed with exit code 1.
FAILED[javac compile][3.0.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.asynchttpclient:async-http-client:2.12.3"]
```

--- a/metadata/org.asynchttpclient/async-http-client/3.0.0/reachability-metadata.json
+++ b/metadata/org.asynchttpclient/async-http-client/3.0.0/reachability-metadata.json
@@ -1,0 +1,36 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.asynchttpclient.netty.channel.ChannelManager"
+      },
+      "type" : "org.asynchttpclient.netty.channel.EpollTransportFactory",
+      "allDeclaredConstructors" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.asynchttpclient.netty.channel.ChannelManager"
+      },
+      "type" : "org.asynchttpclient.netty.channel.KQueueTransportFactory",
+      "allDeclaredConstructors" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.asynchttpclient.filter.ReleasePermitOnComplete"
+      },
+      "type" : {
+        "proxy" : [
+          "org.asynchttpclient.AsyncHandler"
+        ]
+      }
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org.asynchttpclient.config.AsyncHttpClientConfigHelper$Config"
+      },
+      "glob" : "org/asynchttpclient/config/ahc-default.properties"
+    }
+  ]
+}

--- a/metadata/org.asynchttpclient/async-http-client/index.json
+++ b/metadata/org.asynchttpclient/async-http-client/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "3.0.0",
+    "source-code-url" : "https://repo1.maven.org/maven2/org/asynchttpclient/async-http-client/$version$/async-http-client-$version$-sources.jar",
+    "repository-url" : "https://github.com/AsyncHttpClient/async-http-client",
+    "test-code-url" : "https://github.com/AsyncHttpClient/async-http-client/tree/async-http-client-project-$version$/client/src/test",
+    "documentation-url" : "https://repo1.maven.org/maven2/org/asynchttpclient/async-http-client/$version$/async-http-client-$version$-javadoc.jar",
+    "description" : "Async Http Client is a Java library for executing HTTP requests asynchronously and processing HTTP responses without blocking caller threads. It provides a Netty-based client API with support for common HTTP features such as WebSocket, redirects, authentication, proxies, and reactive streams integration.",
     "tested-versions" : [
       "3.0.0"
     ],

--- a/metadata/org.asynchttpclient/async-http-client/index.json
+++ b/metadata/org.asynchttpclient/async-http-client/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.0.0",
+    "tested-versions" : [
+      "3.0.0"
+    ],
+    "allowed-packages" : [
+      "org.asynchttpclient"
+    ]
+  },
+  {
     "metadata-version" : "2.12.3",
     "source-code-url" : "https://repo1.maven.org/maven2/org/asynchttpclient/async-http-client/$version$/async-http-client-$version$-sources.jar",
     "repository-url" : "https://github.com/AsyncHttpClient/async-http-client",

--- a/stats/org.asynchttpclient/async-http-client/3.0.0/execution-metrics.json
+++ b/stats/org.asynchttpclient/async-http-client/3.0.0/execution-metrics.json
@@ -1,0 +1,111 @@
+{
+  "fix_javac_fail:2026-05-04": {
+    "timestamp": "2026-05-04T03:26:10.824362Z",
+    "library": "org.asynchttpclient:async-http-client:3.0.0",
+    "starting_commit": "3332806c07ab6046965f24b2f742ab88dd02bbea",
+    "ending_commit": "b57c4ec20dd391e501a6c1c695b7f8841068d23d",
+    "previous_library": "org.asynchttpclient:async-http-client:2.12.3",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.0.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 5,
+            "totalCalls": 5
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 6,
+        "totalCalls": 6
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 1561,
+          "missed": 27281,
+          "ratio": 0.054122,
+          "total": 28842
+        },
+        "line": {
+          "covered": 385,
+          "missed": 6342,
+          "ratio": 0.057232,
+          "total": 6727
+        },
+        "method": {
+          "covered": 122,
+          "missed": 1581,
+          "ratio": 0.071638,
+          "total": 1703
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.12.3",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 6,
+            "totalCalls": 6
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 7,
+        "totalCalls": 7
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 1662,
+          "missed": 29606,
+          "ratio": 0.053153,
+          "total": 31268
+        },
+        "line": {
+          "covered": 407,
+          "missed": 6704,
+          "ratio": 0.057235,
+          "total": 7111
+        },
+        "method": {
+          "covered": 137,
+          "missed": 1675,
+          "ratio": 0.075607,
+          "total": 1812
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 88376,
+      "cached_input_tokens_used": 771072,
+      "output_tokens_used": 10714,
+      "iterations": 3,
+      "cost_usd": 0.7069,
+      "tested_library_loc": 385,
+      "metadata_entries": 6,
+      "test_only_metadata_entries": 2,
+      "previous_library_metadata_entries": 6,
+      "previous_library_test_only_metadata_entries": 2,
+      "code_coverage_percent": 5.72,
+      "previous_library_coverage_percent": 5.72
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/java/org_asynchttpclient/async_http_client/ReleasePermitOnCompleteTest.java",
+      "metadata_file": "metadata/org.asynchttpclient/async-http-client/3.0.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.asynchttpclient/async-http-client/3.0.0/stats.json
+++ b/stats/org.asynchttpclient/async-http-client/3.0.0/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "3.0.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 5,
+          "totalCalls" : 5
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 6,
+      "totalCalls" : 6
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 1561,
+        "missed" : 27281,
+        "ratio" : 0.054122,
+        "total" : 28842
+      },
+      "line" : {
+        "covered" : 385,
+        "missed" : 6342,
+        "ratio" : 0.057232,
+        "total" : 6727
+      },
+      "method" : {
+        "covered" : 122,
+        "missed" : 1581,
+        "ratio" : 0.071638,
+        "total" : 1703
+      }
+    }
+  } ]
+}

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/.gitignore
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/build.gradle
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.asynchttpclient:async-http-client:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/build.gradle
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/build.gradle
@@ -12,6 +12,7 @@ String libraryVersion = tck.testedLibraryVersion.get()
 
 dependencies {
     testImplementation "org.asynchttpclient:async-http-client:$libraryVersion"
+    testImplementation 'io.netty:netty-transport-classes-epoll:4.1.112.Final'
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/gradle.properties
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.asynchttpclient:async-http-client:3.0.0
+library.version = 3.0.0
+metadata.dir = org.asynchttpclient/async-http-client/3.0.0/

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/settings.gradle
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.asynchttpclient.async-http-client_tests'

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java
@@ -1,0 +1,58 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_asynchttpclient.async_http_client;
+
+import org.asynchttpclient.AsyncHttpClientConfig;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+import org.asynchttpclient.netty.channel.ChannelManager;
+import org.junit.jupiter.api.Test;
+
+import io.netty.util.HashedWheelTimer;
+import io.netty.util.Timer;
+import io.netty.util.concurrent.DefaultThreadFactory;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ChannelManagerTest {
+    @Test
+    void nativeTransportSelectionInstantiatesConfiguredTransportFactory() throws Exception {
+        AsyncHttpClientConfig config = new DefaultAsyncHttpClientConfig.Builder()
+                .setUseNativeTransport(true)
+                .setKeepAlive(false)
+                .setIoThreadsCount(1)
+                .setThreadFactory(new DefaultThreadFactory("ahc-native-transport-test", true))
+                .setShutdownQuietPeriod(0)
+                .setShutdownTimeout(1_000)
+                .build();
+        Timer timer = new HashedWheelTimer(
+                new DefaultThreadFactory("ahc-native-transport-test-timer", true),
+                10,
+                TimeUnit.MILLISECONDS,
+                32);
+        ChannelManager channelManager = null;
+
+        try {
+            channelManager = new ChannelManager(config, timer);
+
+            assertThat(channelManager.getEventLoopGroup().getClass().getName())
+                    .containsAnyOf("epoll", "kqueue");
+        } catch (IllegalArgumentException e) {
+            assertThat(e).hasMessageContaining("No suitable native transport");
+        } finally {
+            if (channelManager != null) {
+                channelManager.close();
+                boolean terminated = channelManager.getEventLoopGroup()
+                        .terminationFuture()
+                        .await(5, TimeUnit.SECONDS);
+                assertThat(terminated).isTrue();
+            }
+            timer.stop();
+        }
+    }
+}

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/java/org_asynchttpclient/async_http_client/ChannelManagerTest.java
@@ -11,15 +11,49 @@ import org.asynchttpclient.DefaultAsyncHttpClientConfig;
 import org.asynchttpclient.netty.channel.ChannelManager;
 import org.junit.jupiter.api.Test;
 
+import io.netty.channel.DefaultEventLoopGroup;
+import io.netty.channel.epoll.EpollEventLoopGroup;
 import io.netty.util.HashedWheelTimer;
 import io.netty.util.Timer;
 import io.netty.util.concurrent.DefaultThreadFactory;
 
+import java.time.Duration;
 import java.util.concurrent.TimeUnit;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
 
 public class ChannelManagerTest {
+    @Test
+    void externalUnknownEventLoopGroupChecksNativeTransportTypes() throws Exception {
+        DefaultEventLoopGroup eventLoopGroup = new DefaultEventLoopGroup(
+                1,
+                new DefaultThreadFactory("ahc-external-event-loop-test", true));
+        AsyncHttpClientConfig config = new DefaultAsyncHttpClientConfig.Builder()
+                .setEventLoopGroup(eventLoopGroup)
+                .setKeepAlive(false)
+                .setThreadFactory(new DefaultThreadFactory("ahc-external-event-loop-config-test", true))
+                .setShutdownQuietPeriod(Duration.ZERO)
+                .setShutdownTimeout(Duration.ofSeconds(1))
+                .build();
+        Timer timer = new HashedWheelTimer(
+                new DefaultThreadFactory("ahc-external-event-loop-test-timer", true),
+                10,
+                TimeUnit.MILLISECONDS,
+                32);
+
+        assertThat(EpollEventLoopGroup.class.getName()).contains("EpollEventLoopGroup");
+
+        try {
+            assertThatExceptionOfType(IllegalArgumentException.class)
+                    .isThrownBy(() -> new ChannelManager(config, timer))
+                    .withMessageContaining("Unknown event loop group DefaultEventLoopGroup");
+        } finally {
+            eventLoopGroup.shutdownGracefully(0, 1, TimeUnit.SECONDS).await(5, TimeUnit.SECONDS);
+            timer.stop();
+        }
+    }
+
     @Test
     void nativeTransportSelectionInstantiatesConfiguredTransportFactory() throws Exception {
         AsyncHttpClientConfig config = new DefaultAsyncHttpClientConfig.Builder()
@@ -27,8 +61,8 @@ public class ChannelManagerTest {
                 .setKeepAlive(false)
                 .setIoThreadsCount(1)
                 .setThreadFactory(new DefaultThreadFactory("ahc-native-transport-test", true))
-                .setShutdownQuietPeriod(0)
-                .setShutdownTimeout(1_000)
+                .setShutdownQuietPeriod(Duration.ZERO)
+                .setShutdownTimeout(Duration.ofSeconds(1))
                 .build();
         Timer timer = new HashedWheelTimer(
                 new DefaultThreadFactory("ahc-native-transport-test-timer", true),

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/java/org_asynchttpclient/async_http_client/NamePasswordCallbackHandlerTest.java
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/java/org_asynchttpclient/async_http_client/NamePasswordCallbackHandlerTest.java
@@ -1,0 +1,37 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_asynchttpclient.async_http_client;
+
+import org.asynchttpclient.spnego.NamePasswordCallbackHandler;
+import org.junit.jupiter.api.Test;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.NameCallback;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class NamePasswordCallbackHandlerTest {
+    @Test
+    void invokesGenericPasswordCallbackMethod() throws Exception {
+        NamePasswordCallbackHandler handler = new NamePasswordCallbackHandler("alice", "secret");
+        NameCallback nameCallback = new NameCallback("User");
+        RecordingPasswordCallback passwordCallback = new RecordingPasswordCallback();
+
+        handler.handle(new Callback[] {nameCallback, passwordCallback});
+
+        assertThat(nameCallback.getName()).isEqualTo("alice");
+        assertThat(passwordCallback.password).containsExactly('s', 'e', 'c', 'r', 'e', 't');
+    }
+
+    public static final class RecordingPasswordCallback implements Callback {
+        private char[] password;
+
+        public void setObject(Object value) {
+            password = (char[]) value;
+        }
+    }
+}

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/java/org_asynchttpclient/async_http_client/ReleasePermitOnCompleteTest.java
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/java/org_asynchttpclient/async_http_client/ReleasePermitOnCompleteTest.java
@@ -1,0 +1,89 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_asynchttpclient.async_http_client;
+
+import org.asynchttpclient.AsyncHandler;
+import org.asynchttpclient.HttpResponseBodyPart;
+import org.asynchttpclient.HttpResponseStatus;
+import org.asynchttpclient.filter.ReleasePermitOnComplete;
+import org.junit.jupiter.api.Test;
+
+import io.netty.handler.codec.http.HttpHeaders;
+
+import java.util.concurrent.Semaphore;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ReleasePermitOnCompleteTest {
+    @Test
+    void releasesPermitAfterSuccessfulCompletion() throws Exception {
+        Semaphore available = new Semaphore(0);
+        RecordingAsyncHandler handler = new RecordingAsyncHandler();
+
+        AsyncHandler<String> wrapped = ReleasePermitOnComplete.wrap(handler, available);
+
+        assertThat(wrapped.onStatusReceived(null)).isEqualTo(AsyncHandler.State.CONTINUE);
+        assertThat(handler.statusCallbacks).isEqualTo(1);
+        assertThat(available.availablePermits()).isZero();
+
+        assertThat(wrapped.onCompleted()).isEqualTo("completed");
+
+        assertThat(handler.completions).isEqualTo(1);
+        assertThat(available.tryAcquire()).isTrue();
+        assertThat(available.availablePermits()).isZero();
+    }
+
+    @Test
+    void releasesPermitAfterThrowableNotification() {
+        Semaphore available = new Semaphore(0);
+        RecordingAsyncHandler handler = new RecordingAsyncHandler();
+        RuntimeException failure = new RuntimeException("request failed");
+
+        AsyncHandler<String> wrapped = ReleasePermitOnComplete.wrap(handler, available);
+        wrapped.onThrowable(failure);
+
+        assertThat(handler.throwables).isEqualTo(1);
+        assertThat(handler.lastThrowable).isSameAs(failure);
+        assertThat(available.tryAcquire()).isTrue();
+        assertThat(available.availablePermits()).isZero();
+    }
+
+    private static final class RecordingAsyncHandler implements AsyncHandler<String> {
+        private int statusCallbacks;
+        private int completions;
+        private int throwables;
+        private Throwable lastThrowable;
+
+        @Override
+        public State onStatusReceived(HttpResponseStatus responseStatus) {
+            statusCallbacks++;
+            return State.CONTINUE;
+        }
+
+        @Override
+        public State onHeadersReceived(HttpHeaders headers) {
+            return State.CONTINUE;
+        }
+
+        @Override
+        public State onBodyPartReceived(HttpResponseBodyPart bodyPart) {
+            return State.CONTINUE;
+        }
+
+        @Override
+        public void onThrowable(Throwable t) {
+            throwables++;
+            lastThrowable = t;
+        }
+
+        @Override
+        public String onCompleted() {
+            completions++;
+            return "completed";
+        }
+    }
+}

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,18 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.asynchttpclient.spnego.NamePasswordCallbackHandler"
+      },
+      "type" : "org_asynchttpclient.async_http_client.NamePasswordCallbackHandlerTest$RecordingPasswordCallback",
+      "methods" : [
+        {
+          "name" : "setObject",
+          "parameterTypes" : [
+            "java.lang.Object"
+          ]
+        }
+      ]
+    }
+  ]
+}

--- a/tests/src/org.asynchttpclient/async-http-client/3.0.0/user-code-filter.json
+++ b/tests/src/org.asynchttpclient/async-http-client/3.0.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.asynchttpclient.**"
+    },
+    {
+      "includeClasses" : "org_asynchttpclient.async_http_client.**"
+    }
+  ]
+}

